### ISSUE TITLE
Fails when installing dart_sass on M1 / Apple Silicon / arm64 Macs

### ISSUE THREAD

**Issue by johnknott** [CargoSense/dart_sass#7]
`M1 Mac - ** (RuntimeError) could not download dart_sass for architecture: aarch64-apple-darwin20.5.0`

It looks like the prebuilt arm64 binaries won't exist until they can be be built on Github Actions - which they can't currently, apparently: 

https://github.com/sass/dart-sass/issues/1125#issuecomment-918635002

What's involved in manually getting this working (natively) on M1 Macs as a stopgap?

I have Erlang (installed through asdf) , Dart and Sass  installed through brew running natively (arm64 - not through Rosetta).

Can I manually copy or symlink the dart and sass binaries into the _build directory?

**Comment by johnknott** [comment]
Works well enough through Rosetta when I manually copy in the Intel binaries, and will fix itself when Dart start shipping arm64 binaries in the future. So I'll close the issue!

**Comment by mcrumm** [comment]
Hi @johnknott, thanks for the issue! We will certainly support the arm64 versions once they're available as releases :)

**Comment by josecfreittas** [comment]
Hi, @mcrumm. Digging [here](https://github.com/sass/dart-sass/issues/1125), it seems that the Sass team is waiting GitHub Actions to support an Apple ARM64 builder, and [here](https://github.com/actions/virtual-environments/issues/2187) we can see that looks like there are no plans yet to GitHub Actions to support it... 

Considering this, and considering that Dart Sass works without problems on the M1 via Rosetta, don't you think a viable solution could be to add a config that allows to override the platform that the binary will be downloaded?

Something like this?

```elixir
config :dart_sass,
  version: "1.39.0",
  default: [
    binary_download_target: "macos-x64",
    args: ~w(css/app.scss ../priv/static/assets/app.css),
    cd: Path.expand("../assets", __DIR__)
  ]
```

**Comment by josecfreittas** [comment]
Hi, @mcrumm. Digging [here](https://github.com/sass/dart-sass/issues/1125), it seems that the Sass team is waiting GitHub Actions to support an Apple ARM64 builder, and [here](https://github.com/actions/virtual-environments/issues/2187) we can see that looks like there are no plans yet to GitHub Actions to support it... 

Considering this, and considering that Dart Sass works without problems on the M1 via Rosetta, don't you think a viable solution could be to add a config that allows to override the platform that the binary will be downloaded?

Something like this?

```elixir
config :dart_sass,
  version: "1.39.0",
  default: [
    binary_download_target: "macos-x64",
    args: ~w(css/app.scss ../priv/static/assets/app.css),
    cd: Path.expand("../assets", __DIR__)
  ]
```

**Comment by clone1018** [comment]
> Works well enough through Rosetta when I manually copy in the Intel binaries, and will fix itself when Dart start shipping arm64 binaries in the future. So I'll close the issue!

Hey John, can you elaborate on how you did this? Do you have to run the entire project via Rosetta, or are you saying that you manually download the libs / executables and copy them somewhere?

**Comment by mcrumm** [comment]
Closed with 75ab0f0
